src/plugins/intel_cpu/src/nodes/reorder.h

```diff
@@ -62,18 +62,105 @@ class Reorder : public Node {
     static std::string getReorderArgs(const MemoryDesc &parentDesc, const MemoryDesc &childDesc);
 
     static void reorderData(const IMemory &input, const IMemory &output, MultiCachePtr cache = nullptr);
+    static void reorderData2(const IMemory &input, const IMemory &output, MultiCachePtr cache = nullptr);
 
 private:
-    dnnl::reorder::primitive prim;
-    std::shared_ptr<MemoryDesc> input;
-    std::shared_ptr<MemoryDesc> output;
-
-    std::vector<int> src_permutation;
-
-    MemoryPtr dst_blocked;
-    MemoryPtr src_blocked;
+    class ReorderExecutor {
```

Comment:
If the executor is supposed to be shared between nodes, it shall have constant internal state. Otherwise we can have unexpected behaviour. The thing is the `prepareParams` method modifies the internal state, but it's not called every time for the other node every infer request since the `Reorder::prepareParams` method is called only when the shape change. Thus, some other node, which got the same executer, may have completelly different memory in the `primArgs` map for example with say nothing about the other parameters, which is of course incorrect.

Reply:
I see! But I thought we have provided - Reorder::reorderData - for other node, and don't need share it to other nodes.
Is it necessary to share it to other nodes?